src/Card/CardSection/index.js

```diff
@@ -1,123 +1,101 @@
-  line-height: ${({ theme }) => theme.orbit.lineHeightTextNormal};
-  color: ${({ theme }) => theme.orbit.colorTextPrimary};
-  border-top: ${({ theme, expanded }) =>
-    expanded
-      ? `1px solid ${theme.orbit.paletteCloudNormal}`
-      : `0px solid ${theme.orbit.paletteCloudNormal}`};
-  padding-top: ${({ theme, expanded }) => expanded && `${theme.orbit.spaceMedium}`};
-  overflow: hidden;
-`;
-
-StyledCardSectionContent.defaultProps = {
-  theme: defaultTheme,
-};
-
-export const StyledCardSection = styled.div`
-  width: 100%;
-  padding: ${({ theme }) => theme.orbit.spaceMedium};
-  box-sizing: border-box;
-  position: relative;
-  background: ${({ theme }) => theme.orbit.backgroundCard};
-
-  ${media.desktop(css`
-    padding: ${({ theme }) => theme.orbit.spaceLarge};
-  `)}
-`;
-
-StyledCardSection.defaultProps = {
-  theme: defaultTheme,
-};
-
-const StyledCardSectionHeader = styled.div`
-  margin-bottom: ${({ theme, expanded }) => expanded && theme.orbit.spaceMedium};
-  display: flex;
-  flex-direction: row;
-  cursor: pointer;
-  position: relative;
-  min-height: ${({ expandable }) => expandable && getSize(ICON_SIZES.MEDIUM)};
-`;
-
-StyledCardSectionHeader.defaultProps = {
-  theme: defaultTheme,
-};
-
-class CardSection extends React.Component<any, Props> {
-  componentDidMount() {
-    const { handleToggleSection, initialExpanded, setInitialExpandedSection } = this.props;
-
-    if (initialExpanded) {
-      handleToggleSection();
-      setInitialExpandedSection();
+const CardSection = ({
+  title,
+  icon,
+  description,
+  children,
+  expandable = false,
+  expanded = false,
+  initialExpanded = false,
+  onClose,
+  header,
+  onExpand,
+  dataTest,
+  actions,
+  noSeparator,
+  dataA11ySection,
+}: Props) => {
+  const [opened, setOpened] = React.useState(expanded || initialExpanded);
+
+  const { onToggle, setExpandedSections, index, roundedBorders, noBorderTop } = useCard();
+
+  React.useEffect(() => {
+    if (expandable && (expanded || initialExpanded)) {
+      setExpandedSections([index]);
     }
-  }
 
-  injectCallbackAndToggleSection = () => {
-    const { handleToggleSection, onClose, onExpand, expanded } = this.props;
-    handleToggleSection(); // First do toggle
+    return () => {
+      setExpandedSections([]);
+    };
+  }, [expandable, expanded, index, initialExpanded, setExpandedSections]);
 
-    if (expanded && onClose) {
-      // If is expanded -> action is closing
+  const handleClick = () => {
+    onToggle();
+
+    if (opened && onClose) {
       onClose();
     }
-    if (!expanded && onExpand) {
-      // if is closed > action is expanding
+    if (!opened && onExpand) {
       onExpand();
     }
+
+    setOpened(state => !state);
   };
 
-  handleKeyDown = (ev: SyntheticKeyboardEvent<HTMLDivElement>) => {
-    if (ev.keyCode === 13 || ev.keyCode === 32) {
-      ev.preventDefault();
+  const handleKeyDown = (ev: SyntheticKeyboardEvent<HTMLDivElement>) => {
+    ev.preventDefault();
```

Comment:
This will prevent all the events on the cardsection so user won't be able to tab through it. Can you prevent only spacebar click? 🙏